fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+    fetch('/api/todos', {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify()
+    })
+      .then(res => res.json())
+      .then(data => this.setState({
+        todos: data
+      }));
+  }
+
+  addTodo(newTodo) {
+    fetch('/api/todos', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newTodo)
+    })
+      .then(res => res.json())
+      .then(data => {
+        this.getAllTodos(data);
+      });
+  }
+
+  toggleCompleted(todoId) {
+    const findTodo = todos => todos.id === todoId;
+    const index = this.state.todos.findIndex(findTodo);
+
+    const complete = this.state.todos[index].isCompleted;
+
+    const checkComplete = {
+      isCompleted: !this.state.isCompleted
+    };
```

Comment:
`this.state.isCompleted` will be undefined so `!this.state.isCompleted` will always be `true`. You should be using your `complete` variable you created above. It should look something like:

```JAVASCRIPT
const checkComplete = {
  isCompleted: !complete
};